src/vtkh/filters/PointAverage.cpp

```diff
@@ -0,0 +1,66 @@
+#include <vtkh/filters/PointAverage.hpp>
+#include <vtkm/filter/PointAverage.h>
+
+namespace vtkh 
+{
+
+PointAverage::PointAverage()
+{
+
+}
+
+PointAverage::~PointAverage()
+{
+
+}
+
+void 
+PointAverage::SetField(const std::string &field_name)
+{
+  m_field_name = field_name;
+}
+
+void 
+PointAverage::SetOutputField(const std::string &field_name)
+{
+  m_output_field_name = field_name;
+}    
+
+void PointAverage::PreExecute() 
+{
+  Filter::PreExecute();
+  assert(m_field_name != "");
+  assert(m_output_field_name != "");
+}
+
+void PointAverage::PostExecute()
+{
+  Filter::PostExecute();
+}
+
+void PointAverage::DoExecute()
+{
+  this->m_output = new DataSet();
+  const int num_domains = this->m_input->GetNumberOfDomains();
+  
+  for(int i = 0; i < num_domains; ++i)
+  {
+    vtkm::Id domain_id;
+    vtkm::cont::DataSet dom;
```

Comment:
At this point you should check to see if this vtkm data set actually has this field, which is not guaranteed. After looking through the other filters, I am not doing this either. I need to make a pass though all the filters and check this in another pull request.